sdcm/stress_thread.py

```diff
@@ -0,0 +1,209 @@
+        elif 'keyspace=' not in stress_cmd:  # if keyspace is defined in the command respect that
+            stress_cmd = stress_cmd.replace(" -schema ", " -schema keyspace=keyspace{} ".format(keyspace_idx))
+
+        cql_auth = self.loader_set.get_cql_auth()
+        if cql_auth and 'user=' not in stress_cmd:
+            # put the credentials into the right place into -mode section
+            stress_cmd = re.sub(r'(-mode.*?)-', r'\1 user={} password={} -'.format(*cql_auth), stress_cmd)
+
+        if self.node_list and '-node' not in stress_cmd:
+            first_node = [n for n in self.node_list if n.dc_idx == loader_idx % 3]  # make sure each loader is targeting on datacenter/region
+            first_node = first_node[0] if first_node else self.node_list[0]
+            stress_cmd += " -node {}".format(first_node.ip_address)
+
+        return stress_cmd
+
+    def _run_stress(self, node, loader_idx, cpu_idx, keyspace_idx):
+        stress_cmd = self.create_stress_cmd(loader_idx, keyspace_idx)
+
+        if self.profile:
+            with open(self.profile) as fp:
+                LOGGER.info('Profile content:\n%s' % fp.read())
+            node.remoter.send_files(self.profile, os.path.join('/tmp', os.path.basename(self.profile)))
+
+        stress_cmd_opt = stress_cmd.split()[1]
+
+        LOGGER.info('Stress command:\n%s' % stress_cmd)
+
+        log_dir = os.path.join(self.output_dir, self.loader_set.name)
+        if not os.path.exists(log_dir):
+            os.makedirs(log_dir)
+        log_file_name = os.path.join(log_dir, 'cassandra-stress-l%s-c%s-k%s-%s.log' % (loader_idx, cpu_idx, keyspace_idx, uuid.uuid4()))
+
+        LOGGER.debug('cassandra-stress local log: %s', log_file_name)
+
+        # This tag will be output in the header of c-stress result,
+        # we parse it to know the loader & cpu info in _parse_cs_summary().
+        tag = 'TAG: loader_idx:%s-cpu_idx:%s-keyspace_idx:%s' % (loader_idx, cpu_idx, keyspace_idx)
+
+        if self.stress_num > 1:
+            node_cmd = 'taskset -c %s bash -c "%s"' % (cpu_idx, stress_cmd)
+        else:
+            node_cmd = stress_cmd
+
+        node_cmd = 'echo %s; %s' % (tag, node_cmd)
+
+        with CassandraStressExporter(instance_name=node.ip_address,
+                                     metrics=nemesis_metrics_obj(),
+                                     cs_operation=stress_cmd_opt,
+                                     cs_log_filename=log_file_name,
+                                     loader_idx=loader_idx, cpu_idx=cpu_idx):
+
+            result = node.remoter.run(cmd=node_cmd,
+                                      timeout=self.timeout,
+                                      ignore_status=True,
+                                      log_file=log_file_name)
+
+        return node, result
+
+    def run(self):
+        if self.round_robin:
+            # cancel stress_num
+            self.stress_num = 1
+            loaders = [self.loader_set.get_loader()]
```

Comment:
check if you can improve with `itertools.cycle`

Reply:
done